### PR TITLE
Add support for MIPS platform and MT7688 boards

### PR DESCRIPTION
**Pull request by sergev**
This patch adds a support for MIPS platform and two boards, based on Mediatek MT7688 chip:
- Onion Omega2 and Omega2+
- Mediatek Linkit Smart 7688 and 7688 Duo

### CODE REVIEW

File: src/mips/mediatek.c [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+    }
+    return MRAA_SUCCESS;
+}
+
+/*
+ * Add a pin descriptor.
+ */
+static void
+add_gpio_pin(mraa_board_t *b, int index, const char *name, int mux,
+    mraa_boolean_t valid, mraa_boolean_t gpio, mraa_boolean_t pwm,
```
Comment (alext-mkrs) [inline src/mips/mediatek.c]: Indentation is not aligned with mraa style.

**Comment by alext-mkrs** [comment]
+general comment - you should really squash your commits and sign the resulting one off (see CONTRIBUTING.MD)

File: docs/linkit_7688.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * Linkit Smart 7688 Duo
+
+
+Pin Mapping
+-----------
+
+| MRAA Number | Physical Pin | Function               |
+|-------------|--------------|------------------------|
+| ---         | P0           | (Reset)                |
+| 43          | P1           | GPIO43 (Eth LED)       |
```
[the remark this answers is not in the document]
Comment (sergev) [reply docs/linkit_7688.md]: Hi Alex,
Thank you for your detailed core review. I'll prepare a new PR with the listed issues fixed. As for pin numbering, it's essentially based on MK7688 datasheet, where the pins have names like GPIO44 with predefined functionality. People already have accustomed to this style and documentation for Linkit and Omega2 boards uses this style. Omega2 board does not have pin designations at all (https://docs.onion.io/omega2-docs/omega2p.html). So I would propose to continue using pin indices as defined by chip spec.

File: src/mips/mediatek.c [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+};
+
+static mraa_result_t
+gpio_init_pre(int pin)
+{
+    struct pinmux *m = &mt7688_mux[gpio_mux_groups[pin]];
+
+    set_gpiomode(m->mask, m->shift, 1);
+
+    return 0;
```
[the remark this answers is not in the document]
Comment (sergev) [reply src/mips/mediatek.c]: It's OK, and `mraa_result_t` works for all functions except `mtk_mmap_read()`, as this function is declared as int in `include/mraa_internal_types.h`:
```
    int (*mmap_read) (mraa_gpio_context dev);
 ```

**Comment by sergev** [comment]
A new pull request created with all issues fixed: #831

File: docs/linkit_7688.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sergev) [reply docs/linkit_7688.md]: After some thinking I came to conclusion that it makes sense to follow the MRAA style and reorder the pins as on boards.

**Pull request closed without merging**